src/student/student.service.ts

```diff
@@ -134,200 +135,164 @@ export class StudentService {
-        return subQuery
-          .select('projects.url', 'bonusProjectUrl')
-          .from(ProjectEntity, 'projects')
-          .where('projects.user_id = :userId', { userId })
-          .andWhere('projects.type="0"');
-      }, 'bonusProjectUrl')
-      .addSelect((subQuery) => {
-        return subQuery
-          .select('projects.url', 'portfolioUrl')
-          .from(ProjectEntity, 'projects')
-          .where('projects.user_id = :userId', { userId })
-          .andWhere('projects.type="1"');
-      }, 'portfolioUrl')
-      .where('student.id = :id', { id })
-      .getRawOne();
-
-    if (!student) {
-      throw new NotFoundException('Kursant z podanym id nie istnieje');
+    async findStudentByUserId(id: string): Promise<StudentEntity> {
+        try {
+            return await this.studentRepository.findOne({where: {userId: id}})
+        } catch {
+            throw new InternalServerErrorException();
+        }
     }
-    console.log(student);
-    return {
-      studentDetails: student,
-    };
-  }
 
-  async updateOne(
-    studentId: string,
-    studentProfileDetails: UpdateStudentDetailsDto,
-  ): Promise<UpdatedStudentResponse> {
-    const student = await this.studentRepository.findOne({
-      where: { id: studentId },
-    });
+    async updateOne(student, studentProfileDetails: UpdateStudentDetailsDto): Promise<UpdatedStudentResponse> {
+        try {
+            const dataUpdatedAt = () => 'CURRENT_TIMESTAMP';
 
-    const dataUpdatedAt = () => 'CURRENT_TIMESTAMP';
+            const {
+                email,
+                bonusProjectUrl,
+                portfolioUrl,
+                githubName,
+                updatedAt,
+                ...restOfDetails
+            } = studentProfileDetails
 
-    const {
-      email,
-      bonusProjectUrl,
-      portfolioUrl,
-      githubName,
-      updatedAt,
-      ...restOfDetails
-    } = studentProfileDetails;
+            const setBonusProjectUrl: UpdateProjectUrlDto = {
+                url: JSON.stringify(bonusProjectUrl),
+                type: projectTypeEnum.bonusProject,
+                updatedAt: dataUpdatedAt,
+            }
+            const setPortfolioUrl: UpdateProjectUrlDto = {
+                url: JSON.stringify(portfolioUrl),
+                type: projectTypeEnum.portfolio,
+                updatedAt: dataUpdatedAt,
+            }
 
-    const setBonusProjectUrl: UpdateProjectUrlDto = {
-      url: JSON.stringify(bonusProjectUrl),
-      type: projectTypeEnum.bonusProject,
-      updatedAt: dataUpdatedAt,
-    };
-    const setPortfolioUrl: UpdateProjectUrlDto = {
-      url: JSON.stringify(portfolioUrl),
-      type: projectTypeEnum.portfolio,
-      updatedAt: dataUpdatedAt,
-    };
+            const githubValidator = await this.githubService.validateGithubName(student.id, githubName)
 
-    if (student) {
-      const userId = await this.getUserByStudentId(studentId);
-      const uniqueUserMail = await this.userRepository.find({
```

Comment:
Nie powinniśmy używać zewnętrznych repozytoriów. Powinniśmy w takiej sytuacji skorzystać bezpośrednio z serwisu w którym to powinna być odpowiednia funkcja. Na tym przykładzie mamy gotową funkcję userService.getByEmail lub userService.getById (jedna z nich wystarczy aby uzyskać interesujący Cię wynik).
Ogólnie w student.service jedyne repozytorium jakiego używamy powinno to być studentRepository a wszelkie inne dane powinny być uzyskiwane z funkcji zewnętrznych serwisów.

Reply:
Ok,  zaczynam to chyba czuć :)